application/sandbox/serial/process/src/services/service.sessions.ts

```diff
@@ -318,6 +403,73 @@ class ServiceSessions {
         });
     }
 
+    private _income_onWriteConfig(message: IPCMessages.PluginInternalMessage): Promise<void> {
+        return new Promise((resolve, reject) => {
+            if (message === undefined) {
+                return reject(new Error(this._logger.error(`Fail to save configuration, because message is undefined`)));
+            }
+            this._updateConfig(message.data.options).then((settings: {}) => {
+                ServiceConfig.write(settings).then(() => {
+                    resolve();
+                }).catch((error: Error) => {
+                    reject(error);
+                });
+            }).catch((error: Error) => {
+                this._logger.error(`Failed to save configurations due error: ${error.message}`);
+                return reject(error);
+            });
+        });
+    }
+
+    private _income_onReadConfig(message: IPCMessages.PluginInternalMessage): Promise<{[port: string]: IOptions}> {
+        return new Promise((resolve, reject) => {
+            ServiceConfig.read<{[key: string]: {[port: string]: IOptions}}>().then(settings => {
+                resolve(settings['ports']);
+            }).catch((error: Error) => {
+                this._logger.error(`Failed to load configurations due error: ${error.message}`);
+                reject(error);
+            });
+        });
+    }
+
+    private _income_onRemoveConfig(message: IPCMessages.PluginInternalMessage): Promise<void> {
+        return new Promise((resolve, reject) => {
+            ServiceConfig.read<{[key: string]: {[port: string]: IOptions}}>().then((settings: {[key: string]: {[port: string]: IOptions}}) => {
+                if (!settings['ports'] || Object.keys(settings['ports']).length === 0) {
+                    return;
+                }
+                if (message.data.port === '*') {
+                    Object.keys(settings['ports']).forEach((port: string) => {
+                        delete settings['ports'][port];
+                    });
+                } else {
+                    delete settings['ports'][message.data.port];
+                }
+                ServiceConfig.write(settings).then(() => {
+                    resolve();
+                }).catch((error: Error) => {
+                    return reject(error);
+                });
+            }).catch((error: Error) => {
+                this._logger.error(`Failed to remove settings due to error: ${error.message}`);
+            });
+        });
+    }
+
+    private _updateConfig(options: IOptions): Promise<{}> {
+        return new Promise((resolve, reject)=> {
+            ServiceConfig.read<{[key: string]: any}>().then((settings: {[key: string]: any}) => {
```

Comment:
See the previous comment. After you will create an interface for settings, you can use it also here `<...>`